tests/integration-tests/tests/cloudwatch_logging/test_cloudwatch_logging.py

```diff
@@ -0,0 +1,461 @@
+            assert_that(next(event_generator, None)).is_not_none()
+
+    def verify_log_group_created(self, log_groups):
+        """Verify whether or not the cluster's log group was created depending on whether it was enabled."""
+        assert_that_log_group_names = assert_that(log_groups).extracting("logGroupName")
+        if self.enabled:
+            assert_that_log_group_names.contains(self.log_group_name)
+        else:
+            assert_that_log_group_names.does_not_contain(self.log_group_name)
+
+    def verify_log_group_retention_days(self, log_groups):
+        """Verify whether or not the cluster's log group was created depending on whether it was enabled."""
+        if not self.enabled:
+            return  # Log group should not be created if not enabled.
+        log_group = next(
+            (group for group in log_groups if group["logGroupName"] == self.log_group_name),
+            None
+        )
+        assert_that(log_group).is_not_none().is_equal_to(
+            {"retentionInDays": self.retention_days}, include="retentionInDays")
+
+    def verify_agent_status(self, logs_state):
+        """Verify CloudWatch agent is running on the MasterServer (or not if not enabled)."""
+        expected_status = "running" if self.enabled else "stopped"
+        assert_that(logs_state).extracting("agent_status").contains_only(expected_status)
+
+    @staticmethod
+    def verify_logs_exist(logs_state):
+        """Verify that the log files expected to exist on the nodes of this cluster do."""
+        for host_dict in logs_state:
+            for log_path, log_dict in host_dict['logs'].items():
+                assert_that(log_dict).is_equal_to({'exists': True}, include='exists')
+
+    def run_tests(self, logs_state):
+        """Run all CloudWatch logging integration tests."""
+        log_groups = self.data_retriever.get_log_groups()
+        self.verify_log_group_created(log_groups)
+        self.verify_log_group_retention_days(log_groups)
+
+        LOGGER.info("state of logs for cluster:\n{}".format(json.dumps(logs_state, indent=4)))
+        self.verify_agent_status(logs_state)
+        self.verify_logs_exist(logs_state)
+
+        if self.enabled:  # Log streams are only relevant when the feature is enabled
+            observed_streams = self.data_retriever.get_log_streams(self.log_group_name)
+            expected_stream_index = self._get_expected_log_stream_index(logs_state)
+            self.verify_log_streams_exist(logs_state, expected_stream_index, observed_streams)
+            self.verify_log_streams_data(logs_state, expected_stream_index, observed_streams)
+
+
+@pytest.mark.parametrize("cw_logging_enabled", [True, False])
+@pytest.mark.regions(["us-west-1", "us-west-2", "us-east-1", "us-east-2"])
+@pytest.mark.instances(["t2.micro", "c4.xlarge"])
+@pytest.mark.oss(["alinux", "centos6", "centos7", "ubuntu1604", "ubuntu1804"])
```

Comment:
remove this - all oss are included by default

Reply:
Done.